.github/workflows/auto-update.js

```diff
@@ -113,29 +119,28 @@ module.exports = async ({ github, context, core, require }) => {
   const url = get_revisions()
     .then(get_newest)
     .then(async (revision) => {
-      const next_sha = revision.sha;
-      const next_url = `https://github.com/nixos/nixpkgs/archive/${next_sha}.tar.gz`;
       const source_path = "./sources.nix";
+      const flake_path = "./flake.nix";
+      const old_flake = readFileSync(flake_path).toString();
+      const next_flake = old_flake.replace(
+        /rev=[a-z0-9]+/,
+        `rev=${revision.sha}`
+      );
 
-      const next_sha256 = await get_sha256(next_url);
+      writeFileSync(flake_path, next_flake);
 
-      const year = revision.date.getFullYear();
-      const month = (revision.date.getMonth() + 1).toString().padStart(2, "0");
-      const day = revision.date.getDate().toString().padStart(2, "0");
+      const [prev_rev, curr_rev] = await update_flake();
 
+      const flake_lock = JSON.parse(readFileSync("./flake.lock").toString());
       const old_source = readFileSync(source_path).toString();
```

Comment:
Same, I guess. I shoulda caught this sooner.
```suggestion
      const old_source = readFileSync(source_path).toString('utf8');
```